frame/referenda/src/types.rs

```diff
@@ -195,6 +195,33 @@ pub struct ReferendumStatus<
 	pub(crate) alarm: Option<(Moment, ScheduleAddress)>,
 }
 
+impl<
+		TrackId: Eq + PartialEq + Debug + Encode + Decode + TypeInfo + Clone,
+		RuntimeOrigin: Eq + PartialEq + Debug + Encode + Decode + TypeInfo + Clone,
+		Moment: Parameter + Eq + PartialEq + Debug + Encode + Decode + TypeInfo + Clone + EncodeLike,
+		Call: Eq + PartialEq + Debug + Encode + Decode + TypeInfo + Clone,
+		Balance: Eq + PartialEq + Debug + Encode + Decode + TypeInfo + Clone,
+		Tally: Eq + PartialEq + Debug + Encode + Decode + TypeInfo + Clone,
+		AccountId: Eq + PartialEq + Debug + Encode + Decode + TypeInfo + Clone,
+		ScheduleAddress: Eq + PartialEq + Debug + Encode + Decode + TypeInfo + Clone,
```

Comment:
This implementation block doesn't require any of these type bounds?

Reply:
It requires the type bounds because the original type is declared with the type bounds 

i.e. one of the errors the compiler returns if we remove the trait bounds:
```
note: required by a bound in `types::ReferendumStatus`
   --> frame/referenda/src/types.rs:171:53
    |
163 | pub struct ReferendumStatus<
    |            ---------------- required by a bound in this
...
171 |     ScheduleAddress: Eq + PartialEq + Debug + Encode + Decode + TypeInfo + Clone,
    |                                                        ^^^^^^ required by this bound in `types::ReferendumStatus`
help: consider restricting type parameter `ScheduleAddress`
    |
206 |         ScheduleAddress: parity_scale_codec::WrapperTypeDecode,
```

Reply:
I removed this code in favor of just making the inner fields pub. That decision should be reviewed.

Reply:
Ahh okay! Ty :)